components/channel_layout/center_channel/center_channel.tsx

```diff
@@ -58,6 +69,27 @@ export default class CenterChannel extends React.PureComponent<Props, State> {
         return {lastReturnTo: nextProps.location.pathname};
     }
 
+    async componentDidMount() {
+        const {actions, showNextSteps, showNextStepsTips, isOnboardingHidden, currentUserId} = this.props;
+        await actions.getProfiles();
+
+        // We call getUser to make sure we have the current user's profile into the profiles state
+        await actions.getUser(currentUserId);
```

Comment:
hey @AGMETEOR just curiosity, why exactly do we need to make this call? like, what can potentially happen if we don't do it?

Reply:
Possibly left over from the previous 5 day creation date check?

Reply:
When we call `getProfiles()` it fetches a paginated list of user profiles. Because the profiles are paginated, it's possible that the page of profiles returned may not have the current user profile. To ensure it's added to the profiles returned we need to make that call `getUser`. It looks redundant but that's how the redux was set up. I'll ask Harisson if they can add to the redux state an entry for the `currentUserProfile`. Atm, there's only `currentUserId`

Reply:
ok, that kind of makes sense. Maybe we can left the note that we should create a getProfileById and just use that one with the currentUserId, internally will be exactly the same, but will make the logic and the code simpler to read.

Reply:
Huh, I'm surprised we don't make a specific call for this at any point during regular operation. At least not that I could find.

Reply:
`getProfiles` will remove the current user from the profiles. See here https://github.com/mattermost/mattermost-webapp/blob/42ac676894524deb94743e3243653f1c9035b201/packages/mattermost-redux/src/actions/users.ts#L304

Reply:
Why do we need to `getProfiles` here in the mount of center channel and how does it relate to the onboarding flow?